### PR TITLE
Update KNX component configuration variable

### PR DESCRIPTION
**Pull request by klaasnicolaas**
Update style of KNX component (all the 7 files) documentation to follow new configuration variables description.
Related to #6385.

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Comment by klaasnicolaas** [comment]
Whoops I got a error: Liquid Exception: (<unknown>): mapping values are not allowed in this context at line 11 column 105 in /opt/build/repo/source/_components/binary_sensor.knx.markdown

File: source/_components/binary_sensor.knx.markdown [hunk 1]
```diff
@@ -57,10 +73,22 @@ binary_sensor:
...
+name:
+  description: A name for this device used within Home Assistant.
+  required: false
+  type: string
+counter:
+  description: Set to 2 if your only want the action to be executed if the button was pressed twice. To 3 for three times button pressed.
+  required: false
+  type: integer
+hook:
+  description: Indicates if the automation should be executed on what state of the binary sensor. Values: "on" or "off".
```
Comment (WofWca) [inline source/_components/binary_sensor.knx.markdown]: `:`

**Review by WofWca — CHANGES_REQUESTED** [review]

File: source/_components/binary_sensor.knx.markdown (hunk 1 above)
Comment (klaasnicolaas) [reply source/_components/binary_sensor.knx.markdown]: Fixed

File: source/_components/cover.knx.markdown [hunk 2]
```diff
@@ -33,17 +33,51 @@ cover:
...
+  type: string
+move_short_address:
+  description: KNX group address for moving the cover short time up or down. If the KNX device has a stop group address you can use that here.
+  required: false
+  type: string
+position_address:
+  description: KNX group address for moving the cover to the dedicated position.
+  required: false
+  type: string
+position_state_address
```
Comment (WofWca) [inline source/_components/cover.knx.markdown]: Missing `:`

**Comment by klaasnicolaas** [comment]
Thnx! Could now be merged?

File: source/_components/binary_sensor.knx.markdown [hunk 3]
```diff
@@ -57,10 +73,22 @@ binary_sensor:
...
-
+{% configuration %}
+name:
+  description: A name for this device used within Home Assistant.
+  required: false
+  type: string
+counter:
+  description: Set to 2 if your only want the action to be executed if the button was pressed twice. To 3 for three times button pressed.
+  required: false
+  type: integer
```
Comment (WofWca) [inline source/_components/binary_sensor.knx.markdown:85]: Default value `1` missing.

File: source/_components/binary_sensor.knx.markdown [hunk 4]
```diff
@@ -57,10 +73,22 @@ binary_sensor:
...
+hook:
+  description: Indicates if the automation should be executed on what state of the binary sensor. Values are "on" or "off".
+  required: false
+  default: on
```
Comment (WofWca) [inline source/_components/binary_sensor.knx.markdown]: This should be quoted. [source](https://github.com/home-assistant/home-assistant/blob/3f87d413813de84935ea67b5212c55348524447f/homeassistant/components/binary_sensor/knx.py#L24).

File: source/_components/cover.knx.markdown [hunk 5]
```diff
@@ -33,17 +33,51 @@ cover:
...
-- **travelling_time_down** (*Optional*): Time cover needs to travel down in seconds. Needed to calculate the intermediate positions of cover while traveling. Defaults to 25.
-- **travelling_time_up** (*Optional*): Time cover needs to travel up in seconds. Needed to calculate the intermediate positions of cover while traveling. Defaults to 25.
-- **invert_position** (*Optional*): Set this to true if your actuator report fully closed as 100%.
-- **invert_angle** (*Optional*): Set this to true if your actuator reports tilt fully closed as 100%.
-
+{% configuration %}
+name:
+  description: A name for this device used within Home Assistant.
+  required: false
+  type: string
```
Comment (WofWca) [inline source/_components/cover.knx.markdown:41]: `name`, `invert_angle`, and `invert_position` actually have defaults too. [Source](https://github.com/home-assistant/home-assistant/blob/3f87d413813de84935ea67b5212c55348524447f/homeassistant/components/cover/knx.py#L35-L49)

File: source/_components/knx.markdown [hunk 6]
```diff
@@ -73,30 +85,51 @@ knx:
...
-- **fire_event_filter** (*Optional*): If `fire_event` is set `fire_event_filter` has to be specified. `fire_event_filter` defines a list of patterns for filtering KNX addresses. Only telegrams which match this pattern are sent to the HOme Assistant event bus. 
-- **state_updater** (*Optional*): The component will collect the current state of each configured device from the KNX bus to display it correctly within Home-Assistant. Set this option to False to prevent this behavior.
+{% configuration %}
+fire_event:
+  description: If set to True, platform will write all received KNX messages to event bus
+  required: false
+  type: boolean
+fire_event_filter:
+  description: If `fire_event` is set `fire_event_filter` has to be specified. `fire_event_filter` defines a list of patterns for filtering KNX addresses. Only telegrams which match this pattern are sent to the HOme Assistant event bus.
+  required: false
```
Comment (WofWca) [inline source/_components/knx.markdown]: [It is inclusive. And it is a list](https://github.com/home-assistant/home-assistant/blob/3f87d413813de84935ea67b5212c55348524447f/homeassistant/components/knx.py#L68).

**Review by WofWca — CHANGES_REQUESTED** [review]
The resulting descriptions mismatch with the source code.

File: source/_components/binary_sensor.knx.markdown (hunk 3 above)
Comment (klaasnicolaas) [reply source/_components/binary_sensor.knx.markdown:85]: Added

File: source/_components/binary_sensor.knx.markdown (hunk 4 above)
Comment (klaasnicolaas) [reply source/_components/binary_sensor.knx.markdown]: Done

File: source/_components/light.knx.markdown [hunk 7]
```diff
@@ -31,16 +31,36 @@ light:
...
-- **state_address** (*Optional*): separate KNX group address for retrieving the switch state of the light.
-- **brightness_state_address** (*Optional*): separate KNX group address for retrieving the dimmed state of the light.
-- **color_address** (*Optional*): separate KNX group address for setting the color of the light.
-- **color_state_address** (*Optional*): separate KNX group address for retrieving the color of the light.
+{% configuration %}
+address:
+  description: KNX group address for switching the light on and off.
+  required: true
+  type: string
+name:
```
Comment (WofWca) [inline source/_components/light.knx.markdown:37]: Optional variables are shown in the config sample.

File: source/_components/sensor.knx.markdown [hunk 8]
```diff
@@ -30,10 +30,21 @@ sensor:
 ```
 
 
+{% configuration %}
+address:
+  description: KNX group address of the sensor.
+  requird: true
```
Comment (WofWca) [inline source/_components/sensor.knx.markdown]: `required`

File: source/_components/sensor.knx.markdown [hunk 9]
```diff
@@ -30,10 +30,21 @@ sensor:
...
+  type: string
+name:
+  description: A name for this device used within Home Assistant.
+  requird: false
```
Comment (WofWca) [inline source/_components/sensor.knx.markdown]: `required`

**Review by WofWca — CHANGES_REQUESTED** [review]

File: source/_components/light.knx.markdown (hunk 7 above)
Comment (klaasnicolaas) [reply source/_components/light.knx.markdown:37]: Have them removed

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks, @klaasnicolaas!

Happy Hacktoberfest 🎉 

../Frenck

**Pull request merged**